compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -1489,6 +1491,34 @@ impl AcirContext {
         let predicate = self.var_to_expression(predicate)?;
         self.acir_ir.brillig(Some(predicate), generated_brillig, b_inputs, b_outputs);
 
+        fn range_constraint_value(
```

Comment:
why is the function defined in the middle of another one?

Reply:
I've done this as it allows us to disregard the `DynamicArray` case. If I turned it into a method on the `AcirContext` we'd need to implement support for `DynamicArray` but here we know it cannot be called with a `DynamicArray`.